flask/flask-common/src/main/java/net/corda/flask/common/LockFile.java

```diff
@@ -24,21 +25,22 @@ private static RandomAccessFile createLockFile(Path path) throws IOException {
     @SneakyThrows
     public static LockFile acquire(Path path, boolean shared) {
         RandomAccessFile raf = createLockFile(path);
-        return new LockFile(raf.getChannel().lock(0L, Long.MAX_VALUE, shared));
+        return new LockFile(raf, raf.getChannel().lock(0L, Long.MAX_VALUE, shared));
     }
 
     @SneakyThrows
     public static LockFile tryAcquire(Path path, boolean shared) {
         RandomAccessFile raf = createLockFile(path);
         FileLock lock = raf.getChannel().tryLock(0L, Long.MAX_VALUE, shared);
         if(lock != null)
-            return new LockFile(lock);
+            return new LockFile(raf, lock);
         else return null;
     }
 
     @Override
     @SneakyThrows
     public void close() {
         lock.release();
+        raf.close();
```

Comment:
Maybe just this is enough?
```java
lock.close()
```
Closing the file descriptor should automatically release all file locks against it. However, file-locking in Java is a wrapper around the platform's native primitives and so you will need to test how this behaves on Windows too.

Reply:
Actually, this looks more likely:
```java
lock.channel().close();
```
Or to be _really_ pedantic:
```java
try {
    lock.close();
} finally {
    lock.channel().close();
}
```

Reply:
so, this was created because of [this](https://r3-cev.atlassian.net/browse/CORE-924)
Thinking about it, I realized we don't really need to use `RandomAccessFile` since we never write to the lockfile, hence I replaced ti with a simple `FileChannel` (and also addressed https://github.com/corda/corda-gradle-plugins/pull/394)